src/matlab2tikz.m

```diff
@@ -5560,6 +5590,66 @@ function warnFacetedInterp(m2t)
     end
 end
 % ==============================================================================
+function [m2t, coords] = makeCoordinates(m2t, varargin)
+    %   [m2t,coords] = makeCoordinates(m2t, 'name1', data1, 'name2', data2, ...)
+    %   [m2t,coords] = makeCoordinates(m2t, {'name1','name2',...}, {data1, data2, ...})
+    %   [m2t,coords] = makeCoordinates(m2t, {'name1','name2',...}, [data1(:), data2(:), ...])
+    %
+    %  Returns m2t structure and formatted coordinate data.
+
+    [variables, data] = parseInputsForTable_(varargin{:});
+
+    COLSEP = sprintf(',');
+    ROWSEP = sprintf('\n');
+
+    nColumns = numel(data);
+    nRows    = cellfun(@numel, data);
+    if ~all(nRows==nRows(1))
+        error('matlab2tikz:makeTableDifferentNumberOfRows',...
+            'Different data lengths [%s].', num2str(nRows));
+    end
+    nRows = nRows(1);
+
+    FORMAT = repmat({m2t.ff}, 1, nColumns);
+    FORMAT(cellfun(@isCellOrChar, data)) = {'%s'};
+    FORMAT(cellfun(@iscategorical, data)) = {'%s'};
+    FORMAT = ['(', join(m2t, FORMAT, COLSEP), ')'];
+
+    for iRow = 1:nRows
+        thisData = cell(1,nColumns);
+        for jCol = 1:nColumns
+            if iscategorical(data{jCol}(iRow))
+                % do not make categorical values lowercase
```

Comment:
This seems to have a lot of duplication with `makeTable`; while both can be merged into a single function, I guess (if not, we should document why). 

I would guess that this part is one of the few places with differences (and picking the `FORMAT` above), so please just adapt `makeTable` to reflect the changes.

I think we can also reduce the complexity of this function quite a bit. E.g. what I notice here is that we can reduce the number of calls to `iscategorical`, but I guess we could even factor out most calls to `lower`, such that we don't even need to check whether the data is categorical in the loop (and for strings, I think we don't need the `lower` per se). But that last one is a note for myself, it is not something you should address.